components/finsh/msh_file.c

```diff
@@ -101,6 +101,7 @@ int msh_exec_script(const char *cmd_line, int size)
         int length;
 
         line_buf = (char *) rt_malloc(RT_CONSOLEBUF_SIZE);
+        if (line_buf == RT_NULL) return -RT_ENOMEM;
```

Comment:
The fd should be closed.

Reply:
Got it, and the code had been updated with your advise.